Adding Azure Pipelines

Nizzan943 opened the pull request:
## Proposed change
Add new service widget - Azure pipelines. 
use to monitor specific pipeline running status and see it buildId.

![image](https://github.com/benphelps/homepage/assets/59150911/4035495b-b8c7-468b-9574-f43b0543b511)
![image](https://github.com/benphelps/homepage/assets/59150911/05020c6d-e2a9-46fd-872d-c64bbcd3aa61)
![image](https://github.com/benphelps/homepage/assets/59150911/fe1c5d75-d374-4414-a7a2-a7c2424c106d)


Closes #1700 

## Type of change

- [X] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (please explain)

## Checklist:

- [X] If adding a service widget or a change that requires it, I have added a corresponding PR to the [documentation](https://github.com/benphelps/homepage-docs) here: [126](https://github.com/benphelps/homepage-docs/pull/126)
- [X] If adding a new widget I have reviewed the [guidelines](https://gethomepage.dev/en/more/development/#service-widget-guidelines).
- [X] If applicable, I have checked that all tests pass with e.g. `pnpm lint`.
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

shamoon commented:
I corrected some syntax issues, please take a look and let me know if you have any concerns

<img width="503" alt="Screenshot 2023-07-20 at 12 02 42 PM" src="https://github.com/benphelps/homepage/assets/4887959/9f8b89f4-4f90-4821-9079-cac5888e9f66">

Nizzan943 commented:
> I corrected some syntax issues, please take a look and let me know if you have any concerns
> 
> <img alt="Screenshot 2023-07-20 at 12 02 42 PM" width="503" src="https://user-images.githubusercontent.com/4887959/254983645-9f8b89f4-4f90-4821-9079-cac5888e9f66.png">

Hey, 
no concerns over here, I modified the represent shape of the build id. show it as number with a comma felt wrong.

I want to make statuses and result string Pascal case, but not sure how to do it in beauty syntax.
make `notStarted` -> `Not Started`
make `succeeded` -> `Succeeded`

cheers!

shamoon commented:
The best way to do that would be with translation strings for each, that would also allow it to be translated into other languages

Nizzan943 commented:
> The best way to do that would be with translation strings for each, that would also allow it to be translated into other languages

Done! thanks for the tip. hope you find it well.
